components/member_list_channel/member_list_channel.tsx

```diff
@@ -63,20 +64,19 @@ export default class MemberListChannel extends React.PureComponent<Props, State>
         };
     }
 
-    componentDidMount() {
+    async componentDidMount() {
         const {
             actions,
             currentChannelId,
             currentTeamId,
         } = this.props;
 
-        actions.loadProfilesAndTeamMembersAndChannelMembers(0, Constants.PROFILE_CHUNK_SIZE, currentTeamId, currentChannelId).then(({data}) => {
-            if (data) {
-                this.loadComplete();
-            }
-        });
-
-        actions.getChannelStats(currentChannelId);
+        await Promise.all([
+            actions.loadProfilesAndTeamMembersAndChannelMembers(0, Constants.PROFILE_CHUNK_SIZE, currentTeamId, currentChannelId),
+            actions.getChannelMembers(currentChannelId),
+            actions.getChannelStats(currentChannelId),
+        ]);
+        this.loadComplete();
```

Comment:
what if something fails? this would still trigger the `loadComplete` call even if something failed.

Reply:
@Willyfrog 
The reason I removed it is because I noticed it was actually not doing anything.
https://github.com/mattermost/mattermost-webapp/blob/master/actions/user_actions.jsx#L51

Notice how the function always returns `data: true`. That's why I removed it.